editor/editor_node.cpp

```diff
@@ -7069,6 +7074,79 @@ void EditorNode::set_unfocused_low_processor_usage_mode_enabled(bool p_enabled)
 	unfocused_low_processor_usage_mode_enabled = p_enabled;
 }
 
+void EditorNode::_update_main_menu_type() {
+	bool use_menu_button = EDITOR_GET("interface/editor/collapse_main_menu");
+	if (use_menu_button) {
+		main_menu_button = memnew(MenuButton);
+		main_menu_button->set_text(TTRC("Main Menu"));
+		main_menu_button->set_theme_type_variation("MainScreenButton");
+		main_menu_button->set_focus_mode(Control::FOCUS_NONE);
+		main_menu_button->set_button_icon(theme->get_icon(SNAME("TripleBar"), EditorStringName(EditorIcons)));
```

Comment:
You could add `is_inside_tree()` check; assigning the icon in the constructor is pointless, because it will get re-assigned anyway.